frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -215,26 +218,28 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
             ? actionItems[0].actionKey
             : _.get(_.keys(sortedList), 0);
       }
-      selectedItem && this.handleChange(selectedItem, sortedList);
+      selectedItem && this.handleChange(selectedItem, sortedList, resourceList);
     }
     return sortedList;
   };
 
-  private handleChange = (key, items) => {
+  private handleChange = (key, items, resourceList) => {
     const name = items[key];
+    const resource = resourceList[key];
+    this.setState({ resources: resourceList });
```

Comment:
This is very odd... you're setting state for something you should have in state unmutated?

Reply:
I have moved this setState out of this function, and made sure the setState is called only when firehose results are loaded or if it has new data.